src/mixins/canvas_events.mixin.js

```diff
@@ -819,12 +819,24 @@
      */
     _fireOverOutEvents: function(target, e) {
       this.fireSyntheticInOutEvents(target, e, {
-        targetName: '_hoveredTarget',
+        oldTarget: this._hoveredTarget,
         canvasEvtOut: 'mouse:out',
         evtOut: 'mouseout',
         canvasEvtIn: 'mouse:over',
         evtIn: 'mouseover',
       });
+      this.targets.forEach(function(_target, index) {
+        this.fireSyntheticInOutEvents(_target, e, {
+          oldTarget: this._hoveredTargets[index],
+          canvasEvtOut: 'mouse:out',
+          evtOut: 'mouseout',
+          canvasEvtIn: 'mouse:over',
+          evtIn: 'mouseover',
+        });
+      });
```

Comment:
here we are missing to fire the out events for all the remaining items in _hoveredTargets that are longer than _targets length.

If _targets is long 2 and _hoveredTargets is long 5, we need to fire 3 out events for the last 3 array positions.
We can do it with another loop or building an array before looping that includes targets and pad up to _hoveredTargets length with nulls.

Reply:
ah, nice catch, i think this is the bug i was experiencing in my first attempt. I will add this in.